marlowe/src/Language/Marlowe/Semantics.hs

```diff
@@ -229,40 +438,34 @@ evalObservation env state obs = let
 
 -- | Pick the first account with money in it
 refundOne :: Map AccountId Money -> Maybe ((Party, Money), Map AccountId Money)
-refundOne accounts = do
-    ((accId, money), rest) <- Map.minViewWithKey accounts
-    if getLovelace money > 0
-    then return ((accountOwner accId, money), rest)
-    else refundOne rest
-
-
-data Payment = Payment Party Money
-  deriving (Eq,Ord,Show,Read)
-
-data ReduceEffect = ReduceWithPayment Payment
-                  | ReduceNoPayment
-  deriving (Eq,Ord,Show,Read)
+refundOne accounts = case Map.toList accounts of
+    [] -> Nothing
+    (accId, balance) : rest ->
+        if balance > 0
+        then Just ((accountOwner accId, balance), Map.fromList rest)
+        else refundOne (Map.fromList rest)
 
 
 -- | Obtains the amount of money available an account
 moneyInAccount :: AccountId -> Map AccountId Money -> Money
-moneyInAccount = Map.findWithDefault (Lovelace 0)
+moneyInAccount accId accounts = case Map.lookup accId accounts of
+    Just x  -> x
+    Nothing -> Ada.lovelaceOf 0
```

Comment:
You can use `mempy` for 0 Ada if you like.

Reply:
Putting just 0 also works, hehe

Reply:
Ah yes, that works for Ada just not for Value.

Reply:
I'm used to saying "use Value everywhere" but I presume multi-currency Marlowe is in the future still :)